### PR TITLE
fix: pinger pod crash on one-shot startup

### LINKED ISSUE #4354: `mirror-1-pinger-…` in CrashLoopBackOff on "successful" deploys
**Reporters:** Matt

**Description**
On at least one deploy reported as successful end-to-end, `kubectl get pods -A` shows the `mirror-1-pinger-...` pod in `CrashLoopBackOff`. Additionally, running solo's start/restart commands shows no pod-status change in k9s.

**Reporter caveat (Matt):**
> Not sure if it was just my setup.

**Suggested action**
- Reproduce on a clean macOS / 32GB box.
- If reproducible, either fix the pinger or remove it from the "ready" criteria.

---

### PR DESCRIPTION
**Pull request by Ivo-Yankov**
## Description

This PR fixes three small issues:

- The Pinger pod crashes when `one-shot` is deployed in parallel mode. This PR fixes it by enabling the Pinger only after the network nodes have started.

- The PR also fixes an issue with `CraneDependencyManager` - on every one-shot command, even when the dependency exists locally, it makes an unnecessary request to the GitHub API, which may result in a 403 error due to GitHub's rate limiting. 

- Removed a wait condition of the Relay during `one-shot`. Since `v0.77.0`, it no longer requires the network node to be running before it gets deployed.

### Related Issues

* Closes #4354 
* Related to #

### Pull request (PR) checklist

* \[ ] This PR added tests (unit, integration, and/or end-to-end)
* \[ ] This PR updated documentation
* \[ ] This PR added no TODOs or commented out code
* \[ ] This PR has no breaking changes
* \[ ] Any technical debt has been documented as a separate issue and linked to this PR
* \[ ] Any `package.json` changes have been explained to and approved by a repository manager
* \[ ] All related issues have been linked to this PR
* \[ ] All changes in this PR are included in the description
* \[ ] When this PR merges the commits will be squashed and the title will be used as the commit message, the 'commit message guidelines' below have been followed

### Testing

* \[ ] This PR added unit tests
* \[ ] This PR added integration/end-to-end tests
* \[ ] These changes required manual testing that is documented below
* \[ ] Anything not tested is documented

The following manual testing was done:

* TBD

The following was not tested:

* TBD

<details>
<summary>
Commit message guidelines
</summary>
We use 'Conventional Commits' to ensure that our commit messages are easy to read, follow a consistent format, and for automated release note generation. Please follow the guidelines below when writing your commit messages:

1. BREAKING CHANGE: a commit that has a footer BREAKING CHANGE:, or appends a ! after the type/scope, introduces a breaking API change (correlating with MAJOR in Semantic Versioning). A BREAKING CHANGE can be part of commits of any type.  NOTE: currently breaking changes will only bump the MAJOR version.
2. The title is prefixed with one of the following:

| Prefix    | Description                                         | Semantic Version Update | Captured in Release Notes |
|-----------|-----------------------------------------------------|-------------------------|---------------------------|
| feat:     | a new feature                                       | MINOR                   | Yes                       |
| fix:      | a bug fix                                           | PATCH                   | Yes                       |
| perf:     | performance                                         | PATCH                   | Yes                       |
| refactor: | code change that isn't feature or fix               | none                    | No                        |
| test:     | adding missing tests                                | none                    | No                        |
| docs:     | changes to documentation                            | none                    | Yes                        |
| build:    | changes to build process                            | none                    | No                        |
| ci:       | changes to CI configuration                         | none                    | No                        |
| style:    | formatting, missing semi-colons, etc                | none                    | No                        |
| chore:    | updating grunt tasks etc; no production code change | none                    | No                        |

</details>

### CODE REVIEW

File: src/core/dependency-managers/base-dependency-manager.ts [hunk 1]
```diff
@@ -279,6 +277,8 @@ export abstract class BaseDependencyManager extends ShellRunner {
       return true;
     }
 
+    await this.preInstall();
+
```
Comment (JeffreyDallas) [inline src/core/dependency-managers/base-dependency-manager.ts:279]: what are the reason or logic move up this preInstall step ?

**Review by JeffreyDallas — DISMISSED** [review]

**Review by jeromy-cannon — DISMISSED** [review]

File: src/core/dependency-managers/base-dependency-manager.ts (hunk 1 above)
Comment (Ivo-Yankov) [reply src/core/dependency-managers/base-dependency-manager.ts:279]: The `preInstall` method in the `craneDependencyManager` sends an API request to GitHub, which is redundant if `isInstalledLocallyAndMeetsRequirements() === true`.

**Review by jan-milenkov — DISMISSED** [review]

**Review by jeromy-cannon — DISMISSED** [review]

**Review by jeromy-cannon — APPROVED** [review]
